ng-dev/release/publish/external-commands.ts

```diff
@@ -37,17 +38,31 @@ import {FatalReleaseActionError} from './actions-error';
  * Invokes the `ng-dev release set-dist-tag` command in order to set the specified
  * NPM dist tag for all packages in the checked out branch to the given version.
  */
-export async function invokeSetNpmDistCommand(npmDistTag: NpmDistTag, version: semver.SemVer) {
+export async function invokeSetNpmDistCommand(
```

Comment:
With 3 yarn commands here, as well as the `npm` commands (which should actually probably be yarn commands) in the release versioning files.  Does it make more sense to have a `YarnClient` similar to our `GitClient` rather than all of this repeated code?

This could also make tests involving the yarn pieces easier potentially.

Reply:
I think there were some issues with publishing through Yarn, so we went with NPM (no idea why though)

Generally I like the idea of keeping it named around "external commands" in this super-dangerous file w/ a file comment. A class could wrap these methods yeah, but I don't think there is any benefit. Yarn would/should need to be re-resolved for every command invocation anyway, and a class has the risk of us caching some logic accidentally (and breaking when we switch branches).

Not sure if that make sense but this just my subjective feeling. This is probably something to revisit in a follow-up regardless?